hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/TopicMetadataLoadingStartupHook.java

```diff
@@ -0,0 +1,132 @@
+import pl.allegro.tech.hermes.common.hook.Hook;
+import pl.allegro.tech.hermes.common.hook.ServiceAwareHook;
+import pl.allegro.tech.hermes.domain.group.GroupRepository;
+import pl.allegro.tech.hermes.domain.topic.TopicRepository;
+import pl.allegro.tech.hermes.frontend.cache.topic.TopicsCache;
+import pl.allegro.tech.hermes.frontend.metric.CachedTopic;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessageProducer;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+public class TopicMetadataLoadingStartupHook implements ServiceAwareHook {
+
+    private static final Logger logger = LoggerFactory.getLogger(TopicMetadataLoadingStartupHook.class);
+
+    private final BrokerMessageProducer brokerMessageProducer;
+
+    private final TopicsCache topicsCache;
+
+    private final GroupRepository groupRepository;
+
+    private final TopicRepository topicRepository;
+
+    private final int retryCount;
+
+    private final long retryInterval;
+
+    @Inject
+    public TopicMetadataLoadingStartupHook(BrokerMessageProducer brokerMessageProducer,
+                                           TopicsCache topicsCache,
+                                           GroupRepository groupRepository,
+                                           TopicRepository topicRepository,
+                                           ConfigFactory config) {
+        this(brokerMessageProducer,
+                topicsCache,
+                groupRepository,
+                topicRepository,
+                config.getIntProperty(Configs.FRONTEND_STARTUP_TOPIC_METADATA_LOADING_RETRY_COUNT),
+                config.getLongProperty(Configs.FRONTEND_STARTUP_TOPIC_METADATA_LOADING_RETRY_INTERVAL));
+    }
+
+    public TopicMetadataLoadingStartupHook(BrokerMessageProducer brokerMessageProducer,
+                                           TopicsCache topicsCache,
+                                           GroupRepository groupRepository,
+                                           TopicRepository topicRepository,
+                                           int retryCount,
+                                           long retryInterval) {
+        this.brokerMessageProducer = brokerMessageProducer;
+        this.topicsCache = topicsCache;
+        this.groupRepository = groupRepository;
+        this.topicRepository = topicRepository;
+        this.retryCount = retryCount;
+        this.retryInterval = retryInterval;
+    }
+
+    @Override
+    public void accept(ServiceLocator serviceLocator) {
+        logger.info("Loading topics metadata");
+        List<TopicName> topicsLeft = getTopics();
+        int initialTopicsCount = topicsLeft.size();
+        int retries = retryCount + 1;
+        while (retries > 0) {
+            retries--;
+            topicsLeft = topicsLeft.stream().filter(this::topicMetadataNotAvailable).collect(toList());
```

Comment:
@dankraw I remember that you wondered about concurrency of this task. Imo we should try to parallel this stream because when we have 500 topics and each metadata download takes 100ms (max is 500ms) then we will wait here 50 seconds which is not short amount of time in startup phase.